fix: kube-proxy.image produces unwanted output

robertoriv opened the pull request:
I was trying out the `kube-proxy` chart and noticed that the `kube-proxy.image` include was prepending the tag value before the repository.

```
❯ cat example.yaml
image:
  repository: myrepository.com/kube-proxy
  tag: v1.0.0

apiServer:
  endpoint: https://myapiserver.com

❯ helm template . -f example.yaml | grep "image:"
          image: :v1.0.0myrepository.com/kube-proxy:v1.0.0
          image: :v1.0.0myrepository.com/kube-proxy:v1.0.0
```

The proposed change retains the check so that one of `tag` or `digest` is set, but doesn't produce the bad output.

Test with tag:
```
❯ cat example.yaml
image:
  repository: myrepository.com/kube-proxy
  tag: v1.0.0

apiServer:
  endpoint: https://myapiserver.com

❯ helm template . -f example.yaml | grep "image:"
          image: myrepository.com/kube-proxy:v1.0.0
          image: myrepository.com/kube-proxy:v1.0.0
```

Test without the tag value:
```
❯ cat example.yaml
image:
  repository: myrepository.com/kube-proxy
  #tag: v1.0.0

apiServer:
  endpoint: https://myapiserver.com

❯ helm template . -f example.yaml | grep "image:"
Error: execution error at (kube-proxy/templates/daemonset.yaml:56:20): At least one of image.tag or image.digest must be provided

Use --debug flag to render out invalid YAML
```

stevehipwell commented on charts/kube-proxy/templates/_helpers.tpl:
```diff
@@ -74,6 +74,8 @@ The image to use
 {{- $digest := ternary (printf "@%s" .Values.image.digest) "" (not (empty .Values.image.digest)) }}
-{{- required "At least one of image.tag or image.digest must be provided" (default $tag $digest) }}
+{{- if and (empty $tag) (empty $digest) }}
+  {{- fail "At least one of image.tag or image.digest must be provided" }}
```
```suggestion
{{- fail "At least one of image.tag or image.digest must be provided" }}
```

stevehipwell reviewed (changes requested):
Thanks for spotting this and providing the fix @robertoriv. Could you please add an entry to the chart CHANGELOG covering your fix. If you want to make the changes for a release that's fine otherwise I'll cut a release once this is merged.

robertoriv commented:
Hi @stevehipwell 

I added the changes to `charts/kube-proxy/CHANGELOG.md`, and updated the version for the chart.

Chart docs have been updated, and your suggested change has been integrated.

stevehipwell reviewed (approved):
Thanks @robertoriv
